_startc.c

```diff
@@ -38,14 +79,10 @@ void _startc(int argc, char **argv, char **env)
 	environ = env;
 	argv_progname = *argv;
 
-	_atexit_init();
-	_errno_init();
-	_malloc_init();
-	_env_init();
-	_signals_init();
-	_file_init();
+	_libc_init();
+
 	_init_array();
-	_pthread_init();
+	atexit(_fini_array);
```

Comment:
not sure if 100% correct - is it allowed for constructors to call `atexit()`? If so - the destructors might not be called last? (probably not regulated by POSIX, but might be relevant for C++ constructors/destructors of global static objects?)

Reply:
I cannot find anything on POSIX about constructors at all, however I see no reaseon not to change the order, thanks for spotting :rocket: